compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -517,6 +517,13 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         )
     }
 
+    /// Get the base address for the bytes in an `Allocation` specified by the
+    /// `AllocID` passed in; error if no such allocation exists.
+    pub fn get_alloc_base_addr(&self, id: AllocId) -> InterpResult<'tcx, usize> {
```

Comment:
```suggestion
    /// Get the base address for the bytes in an `Allocation` specified by the
    /// `AllocID` passed in; error if no such allocation exists. The address will
    /// be exposed (on the host system!).
    ///
    /// It is up to the caller to take sufficient care when using this address:
    /// there could be provenance or uninit memory in there, and other memory
    /// accesses could invalidate the exposed pointer.
    pub fn expose_alloc_base_addr(&self, id: AllocId) -> InterpResult<'tcx, usize> {
```
Also note that this *only allows read-only access to the bytes*. If C code writes to this memory, it's writing to a shared reference! So I think you probably want to make this `mut`.